src/widgets/modals/CashTransactionModal.js

```diff
@@ -333,9 +253,245 @@ export const CashTransactionModal = ({ onConfirm }) => {
+  };
+  const onToggleType = () => dispatch(CashTransactionActions.toggleType());
+  const onUpdateAmount = amount => {
+    dispatch(CashTransactionActions.updateAmount(amount));
+    dispatch(CashTransactionActions.closeInputModal());
+  };
+  const onUpdatePaymentType = paymentType => {
+    dispatch(CashTransactionActions.updatePaymentType(paymentType));
+    dispatch(CashTransactionActions.closeInputModal());
+  };
+  const onUpdateReason = reason => {
+    dispatch(CashTransactionActions.updateReason(reason));
+    dispatch(CashTransactionActions.closeInputModal());
+  };
+  const onUpdateDescription = description => {
+    dispatch(CashTransactionActions.updateDescription(description));
+    dispatch(CashTransactionActions.closeInputModal());
+  };
+  const onInitialiseAmountBuffer = () => dispatch(CashTransactionActions.initialiseAmountBuffer());
+  const onInitialiseDescriptionBuffer = () =>
+    dispatch(CashTransactionActions.initialiseDescriptionBuffer());
+  const onUpdateAmountBuffer = amount => {
+    dispatch(CashTransactionActions.updateAmountBuffer(amount));
+  };
+  const onUpdateDescriptionBuffer = description => {
+    dispatch(CashTransactionActions.updateDescriptionBuffer(description));
+  };
+  const onOpenInputNameModal = () => {
+    dispatch(CashTransactionActions.closeInputModal());
+    dispatch(CashTransactionActions.openInputModal(CASH_TRANSACTION_INPUT_MODAL_FIELDS.NAME));
+  };
+  const onOpenInputAmountModal = () => {
+    dispatch(CashTransactionActions.closeInputModal());
+    dispatch(CashTransactionActions.openInputModal(CASH_TRANSACTION_INPUT_MODAL_FIELDS.AMOUNT));
+  };
+  const onOpenInputPaymentTypeModal = () => {
+    dispatch(CashTransactionActions.closeInputModal());
+    dispatch(
+      CashTransactionActions.openInputModal(CASH_TRANSACTION_INPUT_MODAL_FIELDS.PAYMENT_TYPE)
+    );
+  };
+  const onOpenInputReasonModal = () => {
+    dispatch(CashTransactionActions.closeInputModal());
+    dispatch(CashTransactionActions.openInputModal(CASH_TRANSACTION_INPUT_MODAL_FIELDS.REASON));
+  };
+  const onOpenInputDescriptionModal = () => {
+    dispatch(CashTransactionActions.closeInputModal());
+    dispatch(
+      CashTransactionActions.openInputModal(CASH_TRANSACTION_INPUT_MODAL_FIELDS.DESCRIPTION)
+    );
+  };
+  const onCloseInputModal = () => dispatch(CashTransactionActions.closeInputModal());
+
+  return {
+    onOpen,
+    onClose,
+    onUpdateName,
+    onToggleType,
+    onUpdateAmount,
+    onUpdatePaymentType,
+    onUpdateReason,
+    onUpdateDescription,
+    onInitialiseAmountBuffer,
+    onInitialiseDescriptionBuffer,
+    onUpdateAmountBuffer,
+    onUpdateDescriptionBuffer,
+    onOpenInputNameModal,
+    onOpenInputAmountModal,
+    onOpenInputPaymentTypeModal,
+    onOpenInputReasonModal,
+    onOpenInputDescriptionModal,
+    onCloseInputModal,
+  };
+};
+
+const mapStateToProps = state => {
+  const name = CashTransactionSelectors.name(state);
```

Comment:
Tried something a bit different here, felt like it fit nicely with how we scope actions and reducers, e.g.

```
src/
  actions/
    index.js
    CashTransactionActions.js
  reducers/
    index.js
    CashTransactionReducer.js
  selectors/
    index.js
    CashTransactionSelectors.js
```